src/librvnpal/src/posix/mapping.c

```diff
@@ -50,15 +51,20 @@ rvn_create_and_mmap64_file(
 
     assert(initial_file_size > 0);
 
-    _ensure_path_exists(path);
-      
-    struct map_file_handle* mfh = calloc(1, sizeof(struct map_file_handle));
+    struct map_file_handle *mfh = calloc(1, sizeof(struct map_file_handle));
     *handle = mfh;
     if(mfh == NULL)
     {
         rc = FAIL_CALLOC;
         goto error_cleanup;
     }
+
+    char* directory = strdup(path);
+    int ret = _ensure_path_exists(dirname(directory), detailed_error_code);
```

Comment:
1. verify path != null (however we do allow empty string)
2. dirname will return "." - verify _ensure_path_exists can handle string "."